tests/console/yast2_dns_server.pm

```diff
@@ -141,6 +141,7 @@ sub run {
 
 sub post_fail_hook {
     my $self = shift;
+    select_console 'log-console';
```

Comment:
AFAIR both `console_yasttest` and `y2logsstep` select log-console already. So as per our discussion during planning, could you please try to simply call `$self->SUPER::post_fail_hook()` here instead of `select-console`?

Reply:
the result of the above approach are shown in http://dhcp131.suse.cz/tests/452# . And as you can see it is not working in this case.

Reply:
It actually did work and post-fail hook was triggered, so I'm not sure what do you mean. Looks like it's similar to the issues @jknphy is fixing.

Reply:
yes it did been triggered but something went off and part of the logs from the module are not run. compare the #452 with the VR to see what i mean.

Reply:
http://dhcp131.suse.cz/tests/452#step/yast2_dns_server/448 executes old code, for which I've provided patch on Friday, so could you please rebase and try again?
For other issues, @jknphy  has provided PR, but in general both of your runs end up in the same result, difference is that in one we execute code below first, so problem is less visible.

Reply:
http://dhcp131.suse.cz/tests/452/file/autoinst-log.txt
```
[2019-03-18T10:11:34.812 CET] [debug] /var/lib/openqa/share/tests/sle/tests/console/yast2_dns_server.pm:144 called console_yasttest::post_fail_hook
.
.
.
[2019-03-18T10:27:06.486 CET] [debug] /var/lib/openqa/share/tests/sle/tests/console/yast2_dns_server.pm:144 called console_yasttest::post_fail_hook
[2019-03-18T10:27:06.487 CET] [debug] <<< testapi::wait_serial(regexp=qr/ALEFn-\d+-/, timeout=90)
[2019-03-18T10:28:37.696 CET] [debug] >>> testapi::wait_serial: (?^:ALEFn-\d+-): fail
[2019-03-18T10:28:37.697 CET] [debug] post_fail_hook failed: command 'curl --form upload=@/var/log/zypper.log --form upname=yast2_dns_server-zypper.log http://10.0.2.2:20043/lXVPYwdP08zT3vqE/uploadlog/zypper.log' timed out at /var/lib/openqa/share/tests/sle/lib/console_yasttest.pm line 23.[2019-03-18T10:11:34.812 CET] [debug] /var/lib/openqa/share/tests/sle/tests/console/yast2_dns_server.pm:144 called console_yasttest::post_fail_hook
.
.
.
[2019-03-18T10:27:06.486 CET] [debug] /var/lib/openqa/share/tests/sle/tests/console/yast2_dns_server.pm:144 called console_yasttest::post_fail_hook
[2019-03-18T10:27:06.487 CET] [debug] <<< testapi::wait_serial(regexp=qr/ALEFn-\d+-/, timeout=90)
[2019-03-18T10:28:37.696 CET] [debug] >>> testapi::wait_serial: (?^:ALEFn-\d+-): fail
[2019-03-18T10:28:37.697 CET] [debug] post_fail_hook failed: command 'curl --form upload=@/var/log/zypper.log --form upname=yast2_dns_server-zypper.log http://10.0.2.2:20043/lXVPYwdP08zT3vqE/uploadlog/zypper.log' timed out at /var/lib/openqa/share/tests/sle/lib/console_yasttest.pm line 23
```
It should be fine, however the upload logs issues bothers me as well sometimes.

Reply:
On the otherhand, execution of 3 post_fail_hooks might be a bit over kill and I cannot remember why I have put there multiple inheritance.

Reply:
Or maybe it was there before. Just `console_yasttest` should be enough.

Reply:
@b10n1k if still fails in something strange, try to incorporate manually for a moment this changes in #7053:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7053/files#diff-c51aaa141cc26af0068dbaca9fbfe693R23
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7053/files#diff-870041bc22ca1d13e4d90dd87a13668aR359

Reply:
I followed @rwx788 suggestions and seems to work

Reply:
@b10n1k could you paste your verification with latest change? and I guess it would be ready to merge aftewards.

Reply:
https://openqa.suse.de/tests/2712307